my-app/src/components/search.js

```diff
@@ -0,0 +1,39 @@
+import React, { Component } from 'react';
+
+class Search extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      items : this.props.items,
+      searchitems : this.props.searchitems
+    }
+    this.searchList = this.searchList.bind(this);
+  }
+
+  componentWillReceiveProps(newProps) {
+    this.setState({
+      items:newProps.items,
+      searchitems: newProps.searchitems
+    });
+  }
+
+  searchList(e) {
```

Comment:
Start non-react methods with '_', So it will be distinguishable between react lifecycle methods and custom methods